app.py

```
@@ -1,13 +1,59 @@
-from flask import Flask
+from flask import Flask, jsonify
+import requests
+import os
 
 
 app = Flask(__name__)
 
+base_URL = os.environ.get('BASE_URL',
+                          'https://cplab1.techservices.illinois.edu:443/api')
+
 
 # Canary
 @app.route('/canary')
 def canary():
-    return 'Ok'
+    canary_result = 'Ok'
+    status_code = 200
+
+    # Check Clearpass API dependency
+    token_response = get_clearpass_token(os.environ.get('cp_client_id'),
```

I recommend performing all `os.environ` gets at the start of the file as constants. That way the script/container can fail fast if any necessary environment variables are not given.